packages/api/src/external/carequality/patient.ts

```diff
@@ -23,24 +37,74 @@ export async function discover(
   organization: Organization,
   facilityNPI: string
 ): Promise<void> {
-  let iheGateway: IHEGateway | undefined;
   try {
-    console.log(`IHE Gateway patient discovery - M patientId ${patient.id}`);
-
-    iheGateway = makeIheGatewayAPI();
+    const { id, cxId, data, eTag } = patient;
+    // const iheGateway = makeIheGatewayAPI();
+    console.log(`IHE Gateway patient discovery - M patientId ${id}`);
 
     const fhirPatient = toFHIR(patient);
-
+    const nearbyCQOrgs = await searchNearbyCQOrganizations({
+      cxId,
+      patientId: id,
+    });
+    const xcpdGateways = cqOrgsToXCPDGateways(nearbyCQOrgs);
     const iheGatewayRequest = patientToIheGateway({
       patient: fhirPatient,
       cxId: patient.cxId,
-      xcpdGateways: [], // TODO ADD XCPD GATEWAYS WHEN WE HAVE THEM
+      xcpdGateways,
       facilityNPI: facilityNPI,
       orgName: organization.data.name,
       orgOid: organization.oid,
     });
+    try {
+      // TODO: Uncomment this when testing is done
+      // await pTimeout(
+      //   iheGateway.startPatientDiscovery(iheGatewayRequest),
+      //   PATIENT_DISCOVERY_TIMEOUT.asMilliseconds(),
+      //   "startPatientDiscovery function timed out!"
+      // );
+
+      // TODO: Remove this when testing is done
+      await pTimeout(
+        mockLongExecution(iheGatewayRequest),
+        PATIENT_DISCOVERY_TIMEOUT.asMilliseconds(),
+        "startPatientDiscovery function timed out!"
+      );
+      // Handle successful completion faster than timeout. Think event listeners/emitters?
+    } catch (error) {
+      if (error instanceof pTimeout.TimeoutError) {
+        console.log(error.message);
+      } else {
+        const msg = `Failure while starting patient discovery for ${patient.id} @ IHE Gateway`;
+        console.error(`${msg}. Cause: ${error}`);
+        capture.message(msg, {
+          extra: {
+            facilityNPI,
+            patientId: patient.id,
+            context: createContext,
+            error,
+          },
+          level: "error",
+        });
+      }
+    }
+
+    // At this point, discovery results are stored in the database, so we can retrieve them
+    const discoveryResults = await getPatientDiscoveryResults(iheGatewayRequest.id);
```

Comment:
What's the logic here? Are we always waiting for the timeout, is `iheGateway.startPatientDiscovery` synchronous (as in, returns when the discovery is done)?

I thought the results of PD would go to lambdas